src/jit/morph.cpp

```diff
@@ -11416,6 +11416,20 @@ GenTreePtr Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* mac)
                 }
             }
 
+            // If gtOp1 is a GT_FIELD, we need to pass down the mac if
+            // its parent is GT_ADDR, since the address of the field
+            // is part of an ongoing address computation. Otherwise
+            // op1 represents the value of the field and so any address
+            // calculations it does are in a new context.
+            if ((op1->gtOper == GT_FIELD) && (tree->gtOper != GT_ADDR))
```

Comment:
Yeah what Joe mentioned what I was seeing.  Someone probably Dave Detlefs added this code.
The whole thing feels a bit off to me.  You can see that right before we call the MorphTree on the op1 and op2 we setup this special Context arg and pass it to just the three versions of the Morph function that can use it. 

I guess it does eliminate the extra null check for GT_FIELD expansions but I rather we did this a different way.